packages/apps/plugins/plugin-grid/src/components/Matrix/context.tsx

```diff
@@ -0,0 +1,140 @@
+//
+// Copyright 2024 DXOS.org
+//
+
+import { DetailedCellError, HyperFormula } from 'hyperformula';
+import React, {
+  type CSSProperties,
+  type Dispatch,
+  type PropsWithChildren,
+  type SetStateAction,
+  createContext,
+  useState,
+  useContext,
+  useRef,
+  useEffect,
+} from 'react';
+
+import { Event } from '@dxos/async';
+
+import { type Pos, type Range, rangeToA1Notation, posToA1Notation } from './types';
+
+export type CellEvent = {
+  type: string;
+  pos: Pos;
+  key?: string;
+};
+
+export type MatrixContextType = {
+  event: Event<CellEvent>;
+
+  // Model value.
+  getValue: (pos: Pos) => any;
+  getEditableValue: (pos: Pos) => string;
+  setValue: (pos: Pos, value: any) => void;
+
+  // Current value being edited.
+  text: string;
+  getText: () => string;
+  setText: (text: string) => void;
+
+  // Selection.
+  editing?: Pos;
+  selected?: Range;
+  setSelected: Dispatch<SetStateAction<{ editing?: Pos; selected?: Range }>>;
+
+  outline?: CSSProperties;
+  setOutline: Dispatch<SetStateAction<CSSProperties | undefined>>;
+
+  getDebug: () => any;
+};
+
+const MatrixContext = createContext<MatrixContextType | null>(null);
+
+export const useMatrixContext = (): MatrixContextType => {
+  return useContext(MatrixContext)!;
+};
+
+export const useMatrixEvent = () => {
+  const { event } = useMatrixContext();
+  return event;
+};
+
+export type CellValue = string | number | undefined;
+
+export const MatrixContextProvider = ({ children, data }: PropsWithChildren<{ data?: CellValue[][] }>) => {
+  const [event] = useState(new Event<CellEvent>());
```

Comment:
```suggestion
  const event = useMemo(() => new Event<CellEvent>());
```

Reply:
isn't memo not guaranteed to get called again (thus losing any bound events?)